test/e2e/alerts.js

```diff
@@ -0,0 +1,322 @@
+/* eslint-disable no-console */
+
+import { spawnSync } from 'child_process'
+import { ArgumentParser } from 'argparse'
+import keyBy from 'lodash/keyBy'
+import isMatch from 'lodash/isMatch'
+import pick from 'lodash/pick'
+
+import demoData from '../fixtures/demo.json'
+
+import { createClientInteractive } from 'cozy-client/dist/cli'
+import {
+  ACCOUNT_DOCTYPE,
+  TRANSACTION_DOCTYPE,
+  SETTINGS_DOCTYPE,
+  GROUP_DOCTYPE,
+  BILLS_DOCTYPE
+} from '../../src/doctypes'
+import { importData, getDocumentID } from './dataUtils'
+import { question } from './interactionUtils'
+import Mailhog from 'mailhog'
+
+const LOUISE_ACCOUNT_ID = 'comptelou1'
+const ISA_CHECKING_ACCOUNT_ID = 'compteisa1'
+const ISA_SAVING_ACCOUNT_ID = 'compteisa3'
+
+const demoAccountsById = keyBy(demoData['io.cozy.bank.accounts'], getDocumentID)
+
+const louiseCheckings = demoAccountsById[LOUISE_ACCOUNT_ID]
+const isabelleCheckings = demoAccountsById[ISA_CHECKING_ACCOUNT_ID]
+const isabelleSavings = demoAccountsById[ISA_SAVING_ACCOUNT_ID]
+
+const louiseBurgerTransaction = {
+  demo: false,
+  automaticCategoryId: '400160',
+  account: LOUISE_ACCOUNT_ID,
+  label: 'Burger pour Louise',
+  amount: -8,
+  date: new Date().toISOString(),
+  _id: 'louise_burger'
+}
+
+const isaBurgerTransaction = {
+  ...louiseBurgerTransaction,
+  account: ISA_CHECKING_ACCOUNT_ID,
+  label: 'Burger pour Isabelle',
+  _id: 'isa_burger'
+}
+
+const balanceLower100OnLouiseAccount = {
+  value: 100,
+  accountOrGroup: {
+    _id: LOUISE_ACCOUNT_ID,
+    _type: ACCOUNT_DOCTYPE
+  },
+  enabled: true
+}
+
+const balanceLower300OnAllAccounts = {
+  value: 300,
+  accountOrGroup: null,
+  enabled: true
+}
+
+const settingsWith2BalanceLowerRules = {
+  _id: 'notificationSettings',
+  notifications: {
+    balanceLower: [balanceLower100OnLouiseAccount, balanceLower300OnAllAccounts]
+  }
+}
+
+const fakeTransactionLouise = {
+  account: LOUISE_ACCOUNT_ID,
+  amount: 0,
+  label: 'Fake transaction'
+}
+const fakeTransactionIsa = {
+  account: ISA_CHECKING_ACCOUNT_ID,
+  amount: 0,
+  label: 'Fake transaction'
+}
+
+const scenarios = {
+  balanceLower1: {
+    description: 'Notification for Louise checking account',
+    expectedEmail: {
+      subject: "Balance_alert:_'Louise_checkings'_account_is_at_100€"
```

Comment:
Why do we have underscores in the mail subject?

Reply:
Mailhog returns the subject of emails with underscores. Apparently the rfc2047 stipulates that in email headers, underscores should interpreted as spaces. I will fix this by passing the subject into a replacer. Thanks for noticing, I was going to sweep this under the carpet.